File: debezium/connector/sink/SinkConnectorRunner.java
package com.tiket.tix.hotel.common.debezium.connector.sink;

import com.tiket.tix.hotel.common.debezium.DebeziumHelper;
import com.tiket.tix.hotel.common.debezium.config.DebeziumConfiguration;
import com.tiket.tix.hotel.common.debezium.connector.ConnectorRunner;
import com.tiket.tix.hotel.common.debezium.connector.KafkaConnectorClient;

public class SinkConnectorRunner implements ConnectorRunner {

  private final DebeziumConfiguration config;

  public SinkConnectorRunner(DebeziumConfiguration config) {
    this.config = config;
  }

  @Override
  public void start() {
    try {
      KafkaConnectorClient client = new KafkaConnectorClient();
      String kafkaConnectUrl = config.getKafkaConfiguration().getConnectUrl();

      if (config.getTargetDb() == DebeziumConfiguration.TargetDb.POSTGRES) {
        String json = KafkaSinkConfigBuilder.buildPostgresSink(config);
        client.registerConnector(kafkaConnectUrl, "postgres-sink-connector", json);
        DebeziumHelper.sendMetric("sink_connector_started_use_postgres");

      } else if (config.getTargetDb() == DebeziumConfiguration.TargetDb.MONGO) {
        String json = KafkaSinkConfigBuilder.buildMongoSink(config);
        client.registerConnector(kafkaConnectUrl, "mongo-sink-connector", json);
        DebeziumHelper.sendMetric("sink_connector_started_use_mongo");
      }

    } catch (Exception e) {
      e.printStackTrace();
    }
  }
}

File: debezium/connector/kafka/sink_connector_runner.go
package kafka

import (
	"github.com/tiket/TIX-HOTEL-UTILITIES-GO/debezium/config"
	"github.com/tiket/TIX-HOTEL-UTILITIES-GO/debezium/helper"
)

type SinkConnectorRunner struct {
	cfg *config.DebeziumConfig
}

func NewSinkConnectorRunner(cfg *config.DebeziumConfig) *SinkConnectorRunner {
	return &SinkConnectorRunner{cfg: cfg}
}

func (s *SinkConnectorRunner) Start() error {
	client := NewKafkaConnectorClient()
	kafkaConnectURL := s.cfg.KafkaConfig.ConnectURL

	var jsonConfig string
	switch s.cfg.TargetDB {
	case "POSTGRES":
		jsonConfig = BuildPostgresSink(s.cfg)
		if err := client.RegisterConnector(kafkaConnectURL, "postgres-sink-connector", jsonConfig); err != nil {
			return err
		}
		helper.SendMetric(helper.TagBuilder{
			EventType: "sink_connector_started_use_postgres",
		})
	case "MONGO":
		jsonConfig = BuildMongoSink(s.cfg)
		if err := client.RegisterConnector(kafkaConnectURL, "mongo-sink-connector", jsonConfig); err != nil {
			return err
		}
		helper.SendMetric(helper.TagBuilder{
			EventType: "sink_connector_started_use_mongo",
		})
	}

	return nil
}
